netmaster/objApi/apiController.go

```diff
@@ -0,0 +1,639 @@
+func (ac *APIController) NetworkDelete(network *contivModel.Network) error {
+	log.Infof("Received NetworkDelete: %+v", network)
+
+	// Find the tenant
+	tenant := contivModel.FindTenant(network.TenantName)
+	if tenant == nil {
+		return errors.New("Tenant not found")
+	}
+
+	// Remove link
+	modeldb.RemoveLinkSet(&tenant.LinkSets.Networks, network)
+
+	// Save the tenant too since we added the links
+	err := tenant.Write()
+	if err != nil {
+		return err
+	}
+
+	// Get the state driver
+	stateDriver, err := utils.GetStateDriver()
+	if err != nil {
+		return err
+	}
+
+	// Delete the network
+	err = master.DeleteNetworkID(stateDriver, network.NetworkName)
+	if err != nil {
+		log.Errorf("Error deleting network %s. Err: %v", network.NetworkName, err)
+	}
+
+	return nil
+}
+
+// PolicyCreate creates policy
+func (ac *APIController) PolicyCreate(policy *contivModel.Policy) error {
+	log.Infof("Received PolicyCreate: %+v", policy)
+	return nil
+}
+
+// PolicyUpdate updates policy
+func (ac *APIController) PolicyUpdate(policy, params *contivModel.Policy) error {
+	log.Infof("Received PolicyUpdate: %+v, params: %+v", policy, params)
+	return nil
+}
+
+// PolicyDelete deletes policy
+func (ac *APIController) PolicyDelete(policy *contivModel.Policy) error {
+	log.Infof("Received PolicyDelete: %+v", policy)
+	return nil
+}
+
+// ServiceCreate creates service
+func (ac *APIController) ServiceCreate(service *contivModel.Service) error {
+	log.Infof("Received ServiceCreate: %+v", service)
+
+	// check params
+	if (service.TenantName == "") || (service.AppName == "") {
+		return errors.New("Invalid parameters")
+	}
+
+	// Make sure tenant exists
+	tenant := contivModel.FindTenant(service.TenantName)
+	if tenant == nil {
+		return errors.New("Tenant not found")
+	}
+
+	// Find the app this service belongs to
+	app := contivModel.FindApp(service.TenantName + ":" + service.AppName)
+	if app == nil {
+		return errors.New("App not found")
+	}
+
+	// Setup links
+	modeldb.AddLink(&service.Links.App, app)
+	modeldb.AddLinkSet(&app.LinkSets.Services, service)
+
+	// Save the app too since we added the links
+	err := app.Write()
+	if err != nil {
+		return err
+	}
+
+	// Check if user specified any networks
+	if len(service.Networks) == 0 {
+		service.Networks = append(service.Networks, "privateNet")
+	}
+
+	// link service with network
+	for _, netName := range service.Networks {
+		netKey := service.TenantName + ":" + netName
+		network := contivModel.FindNetwork(netKey)
+		if network == nil {
+			log.Errorf("Service: %s could not find network %s", service.Key, netKey)
+			return errors.New("Network not found")
+		}
+
+		// Link the network
+		modeldb.AddLinkSet(&service.LinkSets.Networks, network)
+		modeldb.AddLinkSet(&network.LinkSets.Services, service)
+
+		// save the network
+		err := network.Write()
+		if err != nil {
+			return err
+		}
+	}
+
+	// Check if user specified any endpoint group for the service
+	if len(service.EndpointGroups) == 0 {
```

Comment:
Is it possible for a service to have fewer endpoint-groups than networks? I am wondering if we need to handle that by creating default-endpoint groups in that case as well.